src/js/modules/setup/controllers/change-password/change-password.ctrl.js

```diff
@@ -0,0 +1,305 @@
+
+    angular.module("blocktrail.setup")
+        .controller("SetupChangePasswordCtrl", SetupChangePasswordCtrl);
+
+    function SetupChangePasswordCtrl($scope, $stateParams, $http, $q, $sce, PasswordStrength, dialogService, $log, $filter, sdkService, $translate, CONFIG) {
+        $scope.working  = false;
+        $scope.error    = null;
+        $scope.form     = {
+            email : null,
+            ERS: null,
+            newPassword: null,
+            newPasswordRepeat: null,
+            passwordCheck: null
+        };
+
+        $scope.walletVersions = ["v1", "v2", "v3"];
+
+        $scope.bip39EN = bip39EN;
+        $scope.stepCount = 0;
+
+        /* Password reset data */
+        var secret = null;
+        var twoFactorToken = null;
+
+        var token = null;
+        var recoverySecret = null;
+        var walletVersion = null;
+        var requires2FA = null;
+
+        // Get state parameters
+        token = $stateParams.token;
+        recoverySecret = $stateParams.recovery;
+        walletVersion = $stateParams.version;
+        requires2FA = $stateParams.requires_2fa;
+        // ----
```

Comment:
no need to seperate this from the `var token = null;, var recoverySecret = null;` etc